Feature/update lohmann

martynpclark opened the pull request:
Mostly modularization

nmizukami commented on route/build/src/read_remap.f90:
```diff
@@ -4,101 +4,115 @@ module read_remap
 private
 
 public::get_remap_data
+public::get_runoff_metadata
```
This routine seems to be out of place. This routine is for runoff data not for remap. maybe move this to read_runoff.f90?

martynpclark commented:
@nmizukami -- I addressed your review comment.

The refactor here is now almost complete. We just need to add the routing code, and these subroutines do not need major changes.
